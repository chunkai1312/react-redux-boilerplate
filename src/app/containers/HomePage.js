import React, { Component, PropTypes } from 'react'
import { connect } from 'react-redux'
import { Counter } from '../components'
import * as actions from '../actions/application'
import * as counter from '../actions/counter'

class HomePage extends Component {

  componentWillMount () {
    this.props.setPageTitle('Home')
  }

  render () {
    const { counter, increment, decrement } = this.props
    return (
      <div>
        <Counter counter={counter} increment={increment} decrement={decrement} />
      </div>
    )
  }
}

HomePage.propTypes = {
  application: PropTypes.object.isRequired,
  counter: PropTypes.number.isRequired,
  setPageTitle: PropTypes.func.isRequired,
  increment: PropTypes.func.isRequired,
  decrement: PropTypes.func.isRequired
}

const mapStateToProps = (state) => ({
  application: state.application,
  counter: state.counter
})

export default connect(mapStateToProps, {
  setPageTitle: actions.setPageTitle,
  increment: counter.increment,
  decrement: counter.decrement
})(HomePage)
