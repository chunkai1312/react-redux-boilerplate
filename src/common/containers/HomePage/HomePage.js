import React, { Component, PropTypes } from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import Helmet from 'react-helmet'
import { MainContent } from '../../components/AppLayout'
import Counter from '../../components/Counter'
import * as actions from '../../actions'

class HomePage extends Component {
  static propTypes = {
    appLayout: PropTypes.object.isRequired,
    counter: PropTypes.number.isRequired,
    actions: PropTypes.object.isRequired
  }

  render () {
    const { counter, actions } = this.props
    return (
      <MainContent>
        <Helmet title="Home" />
        <Counter
          counter={counter}
          increment={actions.increment}
          decrement={actions.decrement}
          incrementAsync={actions.incrementAsync}
        />
      </MainContent>
    )
  }
}

const mapStateToProps = (state) => ({
  appLayout: state.appLayout,
  counter: state.counter
})

const mapDispatchToProps = (dispatch) => ({
  actions: bindActionCreators(actions, dispatch)
})

export default connect(mapStateToProps, mapDispatchToProps)(HomePage)
