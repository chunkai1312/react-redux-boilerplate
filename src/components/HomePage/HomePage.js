import React from 'react'
import Helmet from 'react-helmet'

const HomePage = props => {
  return (
    <div>
      <Helmet title="Home" />Home
    </div>
  )
}

export default HomePage
