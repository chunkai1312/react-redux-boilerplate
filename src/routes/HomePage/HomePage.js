import React, { Component, PropTypes } from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { Counter } from '../../components'
import * as actions from '../../actions'

class HomePage extends Component {
  static propTypes = {
    layout: PropTypes.object.isRequired,
    counter: PropTypes.number.isRequired,
    actions: PropTypes.object.isRequired
  }

  render () {
    const { counter, actions } = this.props
    return (
      <div style={{ padding: '2.4rem' }}>
        <Counter
          counter={counter}
          increment={actions.increment}
          decrement={actions.decrement}
          incrementAsync={actions.incrementAsync}
        />
      </div>
    )
  }
}

const mapStateToProps = (state) => ({
  layout: state.layout,
  counter: state.counter
})

const mapDispatchToProps = (dispatch) => ({
  actions: bindActionCreators(actions, dispatch)
})

export default connect(mapStateToProps, mapDispatchToProps)(HomePage)
