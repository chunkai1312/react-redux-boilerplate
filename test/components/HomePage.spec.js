import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Helmet from 'react-helmet'
import HomePage from '../../src/components/HomePage'

function setup () {
  const wrapper = shallow(<HomePage />)
  return { wrapper }
}

describe('<HomePage />', () => {
  it('should render a <Helmet /> component', () => {
    const { wrapper } = setup()
    expect(wrapper.find(Helmet)).toHaveLength(1)
    expect(wrapper.find(Helmet).node.props.title).toBe('Home')
  })

  it('should render correctly', () => {
    const { wrapper } = setup()
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
