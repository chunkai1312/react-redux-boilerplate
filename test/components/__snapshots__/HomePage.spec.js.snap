// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render correctly 1`] = `
<div>
  <HelmetWrapper
    encodeSpecialCharacters={true}
    title="Home"
  />
  Home
</div>
`;
